lib/controler.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class Controler
+  def initialize(user_interface, actions)
+    @user_interface = user_interface
+    @actions = actions
+  end
+
+  def start
+    actions[user_interface.run.to_i - 1].run
```

Comment:
sugestion, up to you:

now that you have the controller and its become clear why we need an option number (its to access an index of an array) we could make the UI return exactly what we want, **a number**.

So the responsibility for calling `to_i` would be on the user interface, which makes sense, because the UI is in charge of interacting with the exterior and sending the app what it needs, which is why the UI is the one that has the validation logic in place.

Imagine Gandalf shouting "you shall not pass" if it is not the right value. So the UI will make sure it gets an input, that it is the right input, and then will turn it into a number so that the app can use it.